rs/boundary_node/rate_limit_canister/interface.did

```diff
@@ -0,0 +1,114 @@
+type Version = nat64;           // Represents the config version number
+type Timestamp = nat64;         // Represents timestamp in nanoseconds since the epoch (1970-01-01)
+type RuleId = text;             // Unique identifier for each rule
+type SchemaVersion = nat64;     // Version of the schema for encoding/decoding the rules
+type IncidentId = text;         // Unique identifier for each incident
+
+
+// Input structure for defining a rule with mandatory fields within a config
+type InputRule = record {
+  incident_id: IncidentId; // Identifier for the incident, to which the rule is related
+  rule_raw: blob;          // Raw rule data (in binary format), expected to be a valid json object
+  description: text;       // Textual description of the rule
+};
+
+// Output structure for rules
+// Optional fields rule_raw and description may remain hidden while the rule is under confidentiality restrictions
+type OutputRule = record {
+  rule_id: RuleId;                   // Unique identifier for the rule
+  incident_id: IncidentId;           // Identifier for the incident, to which the rule is related
+  rule_raw: opt blob;                // Raw rule data (in binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;             // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;       // Timestamp when the rule was disclosed, none if the rule is still confidential
+};
+
+type OutputConfig = record {
+  schema_version: SchemaVersion; // schema version needed to deserialize the rules
+  rules: vec OutputRule;
```

Comment:
I am not fully sure about this: 
For the API BNs, only `rule_raw` matters. `description`, `disclosed_at`, `incident_id`, `rule_id` only matters for auditing. Maybe we should split the end point into two: one to get the config for the API BNs and there it just returns a vec of rule_raw, while we have a second endpoint that returns a vec of OutputRuleMetadata.

Reply:
I don't have a very strong preference here. Having two functions would lead to code duplication. People inspecting canister code, would need to insure that both functions do the same things.

Reply:
My point is more: Why do we need both `OutputRule` and `OutputRuleMetadata`? This implies that we somehow want to give the basic information about the rules one time and detailed information another time. Then, the question is why is, for example, the `disclosed_at` part of the basic information, but `added_in_version` and `removed_in_version` not.

Reply:
My reasoning was the `OutputRuleMetadata` should be verbose, whereas `OutputRule ` should be more concise.  At the moment they are not very different, i agree. I could either only use `OutputRuleMedata` or shrink  `OutputRule` to:

```
type OutputRule = record {
  rule_id: RuleId; 
  rule_raw: opt blob;
  description: opt text;
};
```

Reply:
left this shortened version of `OutputRule` as agreed.